test/mocks/wrappers/firebase_wrapper_mocks.dart

```diff
@@ -0,0 +1,17 @@
+import 'dart:async';
+
+import 'package:crowdleague/utils/wrappers/firebase_wrapper.dart';
+import 'package:firebase_core/firebase_core.dart';
+
+class FakeFirebaseWrapper implements FirebaseWrapper {
```

Comment:
is this meant to be a mockito fake?

Reply:
thanks mate!  great question!  it wasn't meant to use mockito in that I just wanted to have all the overridden methods there and throw an Unimplemented error, so I just add implements and have the autocorrect fill them out then as the tests break I add a method implementation.  It's more chaos than by design - should it be named differently do you think?  is there a reason to use a mockito fake over a 'real' object that is (as far as I understand) essentially the same thing...?

Reply:
oh by the way I meant to ask if you wanted me to break up the PR, it turned into a bit of a shit show... my 3am coding brain was making some bad calls I think and I threw a bunch of crap in here... I'll go through and add some comments now 🙂

Reply:
and very happy to pull it apart if you'd like, I wanna play around with using cherry picks to split up a PR anyway

Reply:
ooh yeah if your happy to, pulling it apart would help alot 😄

Reply:
and i think naming is fine as is :) and if u want to use all methods then hell yeah we dont need mockito then!

Reply:
shell be right, thanks for the work!